### PR TITLE
Alternative views

### PR DESCRIPTION
**Pull request by encukou**
I'd like to use some representative content in demos. Why not F♥P talking points?

This adds endpoints showing the data as JSON, text, and text with terminal color sequences.

### CODE REVIEW

**Comment by hroncok** [comment]
When freezing the `index.txt` has colorized header while it shouldn't have:

colors


When just serving, I see the following behavior:

 1. start server
 2. fetch txt as a first thing (no color)
 3. fetch term (header no color, rest color)
 4. fetch txt (no color)
 5. keeps repeating

And:

 1. start server
 2. fetch term as a first thing (all color)
 3. fetch txt (header color, rest no color)
 4. fetch term (all color)
 5. keeps repeating

**Comment by hroncok** [comment]
Is there some automagic caching involved?

```python
@app.template_filter('term_color')
def term_color(text, color):
    print(g.colors)
    if g.colors:
        ...
```

```
False printed 22 times
127.0.0.1 - - [07/Aug/2017 16:59:15] "GET /index.txt HTTP/1.1" 200 -
True printed 17 times
127.0.0.1 - - [07/Aug/2017 16:59:24] "GET /index.term HTTP/1.1" 200 -
False printed 17 times
127.0.0.1 - - [07/Aug/2017 17:00:10] "GET /index.txt HTTP/1.1" 200 -
```

**Comment by hroncok** [comment]
Setting `app.jinja_env.cache = None` disables the cache and workarounds the issue.

However, if there is a way to either not use the `g` or mark that part of the template as not cachable or similar, I'd rather use that.

File: templates/index.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+{{ ('='*15) | term_color('blue') }}
+
+{% for point in points %}
+{{ point.name | term_color('white') }}
+{{ point.content | term_indent }}
+
+    {{ point.link.caption | term_color('blue') }} – {{ point.link.href }}
+{% endfor %}
+
+{% filter term_color('white') %}See http://fedoralovespython.org{% endfilter %}
```
Comment (hroncok) [inline templates/index.txt]: One extra newline here makes the response nicer for curl etc.

File: fedoralovespython.py [hunk 2]
```diff
@@ -2,34 +2,81 @@
...
+    return Response(
+        render_template('index.txt', points=points),
+        mimetype='application/octet-stream'
+    )
+
+
+@app.route('/index.json')
+def json():
+    points = read_points()
+    return jsonify(points)
```
Comment (hroncok) [inline fedoralovespython.py:59]: Using `jsonify` directly on points is easy, but it results in:

    "logo": "science"

And that is not quite useful. Maybe we could change the image slug into `url_for('static', filename='img/' + point.logo + '.svg')` in `read_points` and return the URL in JSON and use it directly in the `index.html` template as well?

**Review by hroncok — CHANGES_REQUESTED** [review]
The colors are broken :(

**Comment by encukou** [comment]
Thanks for the review! The jinja caching surprised me.
I pushed fixes that should address your points.

File: templates/index.txt [hunk 3]
```diff
@@ -1,12 +1,14 @@
...
+
 {{ point.content | term_indent }}
 
-    {{ point.link.caption | term_color('blue') }} – {{ point.link.href }}
+    {{ point.link.caption | apply(colors.blue) }} – {{ point.link.href }}
+
 {% endfor %}
 
-{% filter term_color('white') %}See http://fedoralovespython.org{% endfilter %}
+{% filter apply(colors.white) %}See http://fedoralovespython.org{% endfilter %}
```
Comment (hroncok) [inline templates/index.txt]: Actually, could one extra newline be added at the end? Apparently, when there is just one `\n` at the end, that one is not part of the response:

```console
$ curl http://127.0.0.1:8003/index.txt
===============
FEDORA ♥ PYTHON
===============


...



See http://fedoralovespython.org$ 
```

**Comment by encukou** [comment]
Ah, I see what you meant.
That's what I get for testing with less...

**Review by hroncok — APPROVED** [review]
Looks good. Feel free to reorder the commits as desired and merge at will.

**Pull request merged**